crates/miden-objects/src/note/note_tag.rs

```diff
@@ -76,27 +77,33 @@ impl NoteTag {
     // CONSTRUCTORS
     // --------------------------------------------------------------------------------------------
 
-    /// Returns a new [NoteTag] instantiated from the specified account ID and execution mode.
+    /// Returns a new [NoteTag] instantiated from the specified account ID.
     ///
     /// The tag is constructed as follows:
     ///
-    /// - For local execution, the two most significant bits are set to `0b11`, which allows for any
-    ///   note type to be used. The following 14 bits are set to the most significant bits of the
-    ///   account ID, and the remaining 16 bits are set to 0.
-    /// - For network execution, the most significant bits are set to `0b00` and the remaining bits
-    ///   are set to the 30 most significant bits of the account ID.
-    ///
-    /// # Errors
-    ///
-    /// Returns an error if:
-    /// - [`NoteExecutionMode::Network`] is provided but the storage mode of the `account_id` is not
-    ///   [`AccountStorageMode::Network`](crate::account::AccountStorageMode::Network).
-    pub fn from_account_id(
-        account_id: AccountId,
-        execution: NoteExecutionMode,
-    ) -> Result<Self, NoteError> {
-        match execution {
-            NoteExecutionMode::Local => {
+    /// - For local execution(`AccountStorageMode::Private` or `AccountStorageMode::Public`), the
+    ///   two most significant bits are set to `0b11`, which allows for any note type to be used.
+    ///   The following 14 bits are set to the most significant bits of the account ID, and the
+    ///   remaining 16 bits are set to 0.
+    /// - For network execution (`AccountStorageMode::Network`), the most significant bits are set
+    ///   to `0b00` and the remaining bits are set to the 30 most significant bits of the account
+    ///   ID.
+    pub fn from_account_id(account_id: AccountId) -> Result<Self, NoteError> {
```

Comment:
Do we still need to return `Result` from here? It seems like there is no way to get an error in this function - so, probably returning just `Self` should be OK?